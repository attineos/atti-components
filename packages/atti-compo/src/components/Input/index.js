import React from 'react'
import PropTypes from 'prop-types'

import StyledInput from './styles/StyledInput'
import StyledLabelInput from './styles/StyledLabelInput'

/**
 * An input field.
 *
 * @version 0.2
 */
const Input = ({ id, form, label, name, onChange, pattern, placeholder, size, type, value }) => (
  <React.Fragment>
    <StyledLabelInput htmlFor={id}>{label}</StyledLabelInput>
    <StyledInput
      id={id}
      form={form}
      name={name}
      onChange={onChange}
      pattern={pattern}
      placeholder={placeholder}
      size={size}
      type={type}
      value={value}
    />
  </React.Fragment>
)

Input.defaultProps = {
  form: '',
  label: '',
  name: '',
  onChange: undefined,
  pattern: undefined,
  placeholder: '',
  size: null,
  value: undefined,
}

Input.propTypes = {
  /**
   * The id of the input field.
   */
  id: PropTypes.string.isRequired,
  /**
   * One or more forms the Input element belongs to.
   */
  form: PropTypes.oneOfType([PropTypes.string, PropTypes.arrayOf(PropTypes.string)]),
  /**
   * A label associated to the input field.
   */
  label: PropTypes.string,
  /**
   * Name of the input element.
   */
  name: PropTypes.string,
  /**
   * Gets called when the user clicks on the link.
   *
   * @param {SyntheticEvent} event The react `SyntheticEvent`
   */
  onChange: PropTypes.func,
  /**
   * Specifies a regular expression that an Input elemen'ts value is checked against.
   */
  pattern: PropTypes.string,
  /**
   * A short hint that describes the expected value of an Input element.
   */
  placeholder: PropTypes.string,
  /**
   * The width, in characters, of an input element.
   */
  size: PropTypes.number,
  /**
   * The type of the input field.
   */
  type: PropTypes.string.isRequired,
  /**
   * The value of the input element
   */
  value: PropTypes.string,
}

export default Input
