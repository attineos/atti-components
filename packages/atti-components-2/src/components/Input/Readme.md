Normal Input type text :

```js
<Input
  id="inputTestTxt"
  label="Input type text :"
  name="inputTestTxt"
  placeholder="input text"
  type="text"
/>
```

Normal Input type password:

```js
<Input id="inputTest" name="inputTest" placeholder="Password" type="password" />
```
