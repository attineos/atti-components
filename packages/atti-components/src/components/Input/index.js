// @noSnapshot
import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

import StyledInput from './styles/StyledInput'

/**
 * An input field.
 *
 */
const Input = ({
  id,
  form,
  name,
  onChange,
  forwardedRef,
  pattern,
  placeholder,
  size,
  type,
  value,
  variance,
  ...rest
}) => (
  <StyledInput
    form={form}
    id={id}
    name={name}
    onChange={onChange}
    pattern={pattern}
    placeholder={placeholder}
    size={size}
    type={type}
    value={value}
    ref={forwardedRef}
    variance={variance}
    {...rest}
  />
)

Input.defaultProps = {
  form: undefined,
  forwardedRef: undefined,
  id: undefined,
  name: '',
  onChange: undefined,
  pattern: undefined,
  placeholder: '',
  size: null,
  type: 'text',
  value: undefined,
  variance: 'normal',
}

Input.propTypes = {
  /**
   * The id of the input field.
   */
  id: PropTypes.string,
  /**
   * One or more forms the Input element belongs to.
   */
  form: PropTypes.oneOfType([PropTypes.string, PropTypes.arrayOf(PropTypes.string)]),
  /**
   * Name of the input element.
   */
  name: PropTypes.string,
  /**
   * Gets called when the user clicks on the link.
   *
   * @param {SyntheticEvent} event The react `SyntheticEvent`
   */
  onChange: PropTypes.func,
  /**
   * Ref of the input element.
   */
  forwardedRef: PropTypes.func,
  /**
   * Specifies a regular expression that an Input elemen'ts value is checked against.
   */
  pattern: PropTypes.string,
  /**
   * A short hint that describes the expected value of an Input element.
   */
  placeholder: PropTypes.string,
  /**
   * The width, in characters, of an input element.
   */
  size: PropTypes.number,
  /**
   * The type of the input field.
   */
  type: PropTypes.string,
  /**
   * The value of the input element.
   */
  value: PropTypes.any,
  /**
   * Which type of input to display.
   */
  variance: PropTypes.oneOf(['normal', 'error', 'warning']),
}

/** @component */
export default styled(React.forwardRef((props, ref) => <Input {...props} forwardedRef={ref} />))``
