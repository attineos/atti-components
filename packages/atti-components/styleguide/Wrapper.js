import React from 'react'

import ThemeProvider from '../src/components/ThemeProvider'
import { defaultTheme } from '../src/themes'

export default ({ children }) => <ThemeProvider otherTheme={defaultTheme}>{children}</ThemeProvider>
