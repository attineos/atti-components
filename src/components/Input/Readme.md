Normal Input type text :

```js
initialState = { value: '' }
onChangeFunction = e => {
  setState({ value: e.target.value })
}
;<div>
  <p>Value Selected : {state.value}</p>
  <p>
    <Input
      id="inputTestTxt"
      label="Input type text :"
      name="inputTestTxt"
      onChange={onChangeFunction}
      placeholder="input text"
      type="text"
      value={state.value}
    />
  </p>
</div>
```

Normal Input type password:

```js
<Input id="inputTest" name="inputTest" placeholder="Password" type="password" />
```

Normal Input type tel:

```js
<Input id="inputTest" name="inputTestTel" type="tel" />
```
