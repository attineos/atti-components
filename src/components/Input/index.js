import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

import StyledInput from './styles/StyledInput'

/**
 * An input field.
 *
 */
class Input extends PureComponent {
  render() {
    const {
      className,
      id,
      form,
      name,
      onChange,
      _ref,
      pattern,
      placeholder,
      size,
      type,
      value,
      variance,
      ...rest
    } = this.props

    return (
      <StyledInput
        className={className}
        form={form}
        id={id}
        name={name}
        onChange={onChange}
        pattern={pattern}
        placeholder={placeholder}
        size={size}
        type={type}
        value={value}
        ref={_ref}
        variance={variance}
        {...rest}
      />
    )
  }
}

Input.defaultProps = {
  className: '',
  form: undefined,
  name: '',
  onChange: undefined,
  _ref: undefined,
  pattern: undefined,
  placeholder: '',
  size: null,
  value: undefined,
  variance: 'normal',
}

Input.propTypes = {
  /**
   * The classes of the input field.
   */
  className: PropTypes.string,
  /**
   * The id of the input field.
   */
  id: PropTypes.string.isRequired,
  /**
   * One or more forms the Input element belongs to.
   */
  form: PropTypes.oneOfType([PropTypes.string, PropTypes.arrayOf(PropTypes.string)]),
  /**
   * Name of the input element.
   */
  name: PropTypes.string,
  /**
   * Gets called when the user clicks on the link.
   *
   * @param {SyntheticEvent} event The react `SyntheticEvent`
   */
  onChange: PropTypes.func,
  /**
   * Ref of the input element.
   */
  _ref: PropTypes.func,
  /**
   * Specifies a regular expression that an Input elemen'ts value is checked against.
   */
  pattern: PropTypes.string,
  /**
   * A short hint that describes the expected value of an Input element.
   */
  placeholder: PropTypes.string,
  /**
   * The width, in characters, of an input element.
   */
  size: PropTypes.number,
  /**
   * The type of the input field.
   */
  type: PropTypes.string.isRequired,
  /**
   * The value of the input element
   */
  value: PropTypes.string,
  /**
   * Which type of input to display
   */
  variance: PropTypes.oneOf(['normal', 'error', 'warning']),
}

/** @component */
export default styled(Input)``
