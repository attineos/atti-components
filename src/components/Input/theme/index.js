export default ({ forms, spaces }) => ({
  input: {
    borders: {
      radius: forms.borders.radius,
      width: forms.borders.width,
    },
    boxShadows: {
      boxShadowFocus: forms.boxShadow,
    },
    colors: {
      border: forms.colors.border,
      borderFocus: forms.colors.borderFocus,
      caret: forms.colors.caret,
      placeholder: forms.colors.placeholder,
      text: forms.colors.text,
    },
    sizes: {
      minHeight: forms.sizes.minHeight,
    },
    spaces: {
      paddingWidth: spaces.s5,
    },
  },
})
