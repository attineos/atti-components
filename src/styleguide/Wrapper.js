import React from 'react'
import PropTypes from 'prop-types'
import { remove, get } from 'lodash'

import Cookies from './helpers/Cookies'
import themes from './config/themes'

import ThemeProvider from '../components/ThemeProvider'

const updateWrappers = []

export const getWrappersToUpdate = () => updateWrappers

class Wrapper extends React.Component {
  static propTypes = {
    /**
     * The child node.
     */
    children: PropTypes.node.isRequired,
  }

  state = {
    selectedTheme: null,
  }

  componentWillMount() {
    updateWrappers.push(this)

    const selectedTheme = Cookies.hasItem('selectedTheme')
      ? Cookies.getItem('selectedTheme')
      : 'default'

    this.state.selectedTheme = get(themes, selectedTheme, null)
  }

  componentWillUnmount() {
    remove(updateWrappers, item => item === this)
  }

  handleSetTheme = newTheme =>
    this.setState(
      {
        selectedTheme: newTheme,
      },
      () => this.forceUpdate(),
    )

  render() {
    const { children } = this.props
    const { selectedTheme } = this.state

    return (
      <ThemeProvider theme={selectedTheme} key={Math.random()}>
        {children}
      </ThemeProvider>
    )
  }
}

export default Wrapper
