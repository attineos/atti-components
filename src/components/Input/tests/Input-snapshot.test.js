/* This file was generated automatically, please do not write in it. */
import React from 'react';
import Input from '..';
import Wrapper from 'styleguide/Wrapper'
import renderer from 'react-test-renderer';

const mock = () => {}

it('Input renders correctly', () => {
  const tree = renderer
    .create(
      <Wrapper>
        <Input id="Atticomponents" form={true} label="Atticomponents" name="Atticomponents" onChange={mock} pattern="Atticomponents" placeholder="Atticomponents" size={42} type="Atticomponents" value="Atticomponents" ></Input>
      </Wrapper>
      )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
    