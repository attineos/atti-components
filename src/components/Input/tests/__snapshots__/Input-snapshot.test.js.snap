// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders correctly 1`] = `
Array [
  <label
    className="sc-bwzfXH hhqXtE"
    htmlFor="Atticomponents"
  >
    Atticomponents
  </label>,
  <input
    className="sc-bdVaJa hUvXGl"
    form={true}
    id="Atticomponents"
    name="Atticomponents"
    onChange={[Function]}
    pattern="Atticomponents"
    placeholder="Atticomponents"
    size={42}
    type="Atticomponents"
    value="Atticomponents"
  />,
]
`;
